service/src/main/java/org/broadinstitute/listener/relay/transport/RelayedRequestPipeline.java

```diff
@@ -92,16 +91,12 @@ public void registerHttpExecutionPipeline(Scheduler scheduler) {
                             return httpRequestProcessor.executeRequestOnTarget(c);
                           }
                           httpRequestProcessor.writeNotAcceptedResponseOnCaller(c);
-
-                          return Mono.empty();
+                          return null;
```

Comment:
Before this was a `Mono` of a `Mono.empty`, causing the `ClassCastException` down below. I think `Mono.fromCallable(() -> null)` resolves to `Mono.empty`.

Reply:
In general is not a good practice to return null in a reactor pipeline.  Instead I would keep the return `Mono.empty()` and then use operator `.switchIfEmpty` before calling to other map, and indicate an error, something like this:
```java
 .switchIfEmpty(Mono.Error(new RuntimeException("call was not accepted by the inspectors")))
 .flatMap(
   (r) -> Mono.fromCallable(
                        () ->
                            httpRequestProcessor.writeTargetResponseOnCaller(
                                (TargetHttpResponse) r))
```

Reply:
@giventocode fair point about null but I don't think the above code resolves the issue? I don't think `switchIfEmpty` will fire since it's not empty -- it's a `Mono.fromCallable(() -> Mono.empty())`. At least I still get the `ClassCastException` from `(TargetHttpResponse) r` with the above code.

Thinking about other ways to do it..

Reply:
hmm I see, you are correct. The following change seems to fix the problem - the test passed without the class exception thrown/logged.
```java
.flatMap(
            (c) ->
                Mono.fromCallable(
                        () -> {
                          if (listenerConnectionHandler.isRelayedHttpRequestAcceptedByInspectors(
                              c.getRequest())) {
                            return httpRequestProcessor.executeRequestOnTarget(c);
                          }
                          httpRequestProcessor.writeNotAcceptedResponseOnCaller(c);
                          return Mono.empty();
                        })
                    .subscribeOn(scheduler))
        .flatMap(
            (r) ->
                Mono.fromCallable(
                        () -> {
                          if (r instanceof TargetHttpResponse) {
                            return httpRequestProcessor.writeTargetResponseOnCaller(
                                (TargetHttpResponse) r);
                          }
                          return Result.FAILURE;
                        })
                    .subscribeOn(scheduler))
                    ```